src/mlx_omni_server/chat/mlx/tools/reasoning_decoder.py

```
@@ -0,0 +1,130 @@
+        # Check if accumulated text already contains the end tag
+        has_end_tag_before = thinking_end_tag in self.accumulated_text[: -len(text)]
+        has_end_tag_now = thinking_end_tag in self.accumulated_text
+
+        # If text starts with thinking tag and end tag hasn't been encountered yet
+        if self.accumulated_text.startswith(thinking_start_tag) and not has_end_tag_now:
+            # delta content as reasoning
+            return {"delta_content": None, "delta_reasoning": text}
+        # If current delta contains the end tag (from not having it to having it)
+        elif not has_end_tag_before and has_end_tag_now:
+            # Split the current delta
+            parts = text.split(thinking_end_tag, 1)
```

Ruff (0.11.9)

Local variable `parts` is assigned to but never used

Remove assignment to unused variable `parts`

(F841)